src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj

```diff
@@ -19,6 +19,12 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU' "/>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU' "/>
+  <PropertyGroup>
+    <TargetsBSD Condition="'$(TargetsOSX)' == 'true' or '$(TargetsFreeBSD)' == 'true'">true</TargetsBSD>
+  </PropertyGroup>
```

Comment:
> Right now it is a big mess [...] Like this it just makes absolutely no sense.

Really? I do appreciate your point of view (and your contributions and insights), and I'm happy to have these kinds of discussions.  I'd like it if we could avoid exaggeration and hyperbole, though; at least that's how I interpret these statements.  Apologies if I've misconstrued them.

> Also if we go for APIs, we must not have TargetOSX, TargetLinux etc. Then you can only define TargetsPOSIX2008, TargetsBSD4.4, TargetsWin32, etc.

We could very well find that we need such specificity and level of categorization in the future; if and when we do, we can introduce it... the code base is malleable in that regard, and we're not locked into any one approach (I expect we'll see some changes here related to #2137, mostly around consolidating things which we've had to split apart).  But thus far we haven't seen such a need, and I don't believe we're currently in a place that would benefit from it.  While I agree that if we were to apply pure academic rigor such an approach could be warranted, perfection can be the enemy of the good, and I believe where we are strikes a nice balance.  If that changes as the code base grows, we'll all adapt with it.

> I would really appreciate if we could get past bikeshedding this single issue in a otherwise most excellent and productive patch.

Completely agree.  Jon should go ahead with fixing up the other issues in the PR and getting it checked in.